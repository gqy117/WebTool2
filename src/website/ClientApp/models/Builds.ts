export interface BuildConfigration {
    buildId: string;
    refreshInverval?: number ;
}

export interface BuildViewDTO {
    view?: BuildView;
}

export interface BuildView {
    buildId?: string;
    buildStatus?: string;
    buildName?: string;
    lastUpdated?: string;
    cssStatus: string;
    imageSrc: string;
}

export interface BuildState {
    buildState?: Build[];
}

export interface Build {
    buildId?: string;
    buildStatus?: string;
    buildName?: string;
    lastUpdated?: string;
}

export const enum ActionType {
    REQUEST_BUILD = "REQUEST_BUILD",
    RECEIVE_BUILD = "RECEIVE_BUILD",
}

export const enum BuildStatus {
    SUCCESSFUL = "Successful",
    FAILED = "Failed",
}

export const enum CssStatus {
    SUCCESSFUL = "status-successful",
    WARNING = "status-warning",
}

export interface RequestBuildsAction {
    type: ActionType.REQUEST_BUILD;
}

export interface ReceiveBuildsAction {
    type: ActionType.RECEIVE_BUILD;
    build: Build;
}

export type KnownAction = RequestBuildsAction | ReceiveBuildsAction;
