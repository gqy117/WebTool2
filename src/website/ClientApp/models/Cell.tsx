import { RowRenderProps } from "react-table";

export interface CellProps {
    className?: string;
    rowRenderProps: RowRenderProps;
}
