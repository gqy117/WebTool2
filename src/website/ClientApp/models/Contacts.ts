import { ControlledStateOverrideProps, } from "react-table";
import { Dictionary } from "../utilities";
export { ControlledStateOverrideProps, } from "react-table";

export interface ContactQuery {
    isFetching?: boolean;
    name?: string;
    gender?: string;
    address?: string;
    birthdayFrom?: string;
    birthdayTo?: string;
    phone?: string;
    paging?: ControlledStateOverrideProps;
}

export interface ContactState {
    query?: ContactQuery;
    contactState?: ContactResultSet;
}

export interface ContactResultSet {
    contacts?: Contact[];
    paging?: ControlledStateOverrideProps;
}

export interface Contact {
    name: string;
    address: string;
}

export const enum ActionType {
    QUERY_CHANGED = "QUERY_CHANGED",
    REQUEST_CONTACT = "REQUEST_CONTACT",
    RECEIVE_CONTACT = "RECEIVE_CONTACT",
}

export interface QueryChangedAction {
    type: ActionType.QUERY_CHANGED;
    name: string;
    value: string;
}

export interface RequestContactsAction {
    type: ActionType.REQUEST_CONTACT;
    contactQuery: ContactQuery;
}

export interface ReceiveContactsAction {
    type: ActionType.RECEIVE_CONTACT;
    contactResultSet: ContactResultSet;
}

export const enum Gender {
    Default = "",
    M = "M",
    F = "F",
}

export const enum GenderKey {
    All = "0",
    Male = "1",
    Female = "2",
}

export const enum GenderValue {
    All = "All",
    Male = "Male",
    Female = "Female",
}

export const enum GenderCss {
    Default = "",
    M = "male",
    F = "female",
}

export const GenderCssDic = new Dictionary<string>([
    { k: Gender.Default, v: GenderCss.Default },
    { k: Gender.M, v: GenderCss.M },
    { k: Gender.F, v: GenderCss.F },
]);

export const GenderSelectorDic = new Dictionary<string>([
    { k: GenderKey.All, v: GenderValue.All },
    { k: GenderKey.Male, v: GenderValue.Male },
    { k: GenderKey.Female, v: GenderValue.Female },
]);

export type DefaultCellProps = ContactQuery;
