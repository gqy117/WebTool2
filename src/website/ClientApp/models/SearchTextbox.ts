export interface SearchText {
    id: string;
    className?: string;
    iconClassName?: string;
    iconDivClassName?: string;
    fieldName: string;
    fieldValue?: string;
    label: string;
    placeholder: string;
    isLast?: boolean;
    tooltip?: JSX.Element;
}
