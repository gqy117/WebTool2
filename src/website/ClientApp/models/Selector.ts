export interface Selector {
    id: string;
    fieldName: string;
    fieldValue?: string;
}
