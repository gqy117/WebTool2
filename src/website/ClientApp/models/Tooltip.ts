export interface TooltipProps {
    overlay?: string;
}
