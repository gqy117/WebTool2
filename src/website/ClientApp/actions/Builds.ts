import { fetch, addTask } from "domain-task/index";
import { AppThunkAction } from "../reducers";
import * as Model from "../models/Builds";

export const actionCreators = {
    requestBuilds: (buildId?: string): AppThunkAction<Model.KnownAction> => (dispatch) => {
        let fetchTask = fetch(`api/TeamCity/Build?buildId=${buildId}`)
            .then(response => response.json() as Promise<Model.Build>)
            .then(data => {
                //console.log(data);
                dispatch({ type: Model.ActionType.RECEIVE_BUILD, build: data });
            });

        addTask(fetchTask);
        dispatch({ type: Model.ActionType.REQUEST_BUILD });
    }
};