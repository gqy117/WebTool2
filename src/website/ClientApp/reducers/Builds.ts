import { Reducer } from "redux/index";
import * as Model from "../models/Builds";

const initialState: Model.BuildState = { buildState: [] };

export const reducer: Reducer<Model.BuildState> = (state: Model.BuildState = initialState, action: Model.KnownAction) => {
    switch (action.type) {
        case Model.ActionType.RECEIVE_BUILD:
            const buildWithoutCurrent = (state.buildState as Model.Build[]).filter((x) => x.buildId !== action.build.buildId);

            state = {
                buildState: [
                    ...buildWithoutCurrent,
                    action.build,
                ]
            };
            break;

        case Model.ActionType.REQUEST_BUILD:
        default:
            break;
    }
    // console.log(state);
    return state;
};
