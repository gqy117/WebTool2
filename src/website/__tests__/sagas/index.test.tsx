import * as Model from "../../ClientApp/models/Builds";
import rootSaga from "../../ClientApp/sagas";
import * as Const from "../consts/Builds";

describe("sagas/index", () => {
    it("rootSaga", () => {
        const gen = rootSaga();

        expect(gen.next().value).toMatchSnapshot();
        expect(gen.next(Const.DCE_MAIN_SUCCESSFUL).value).toMatchSnapshot();
    });

});
