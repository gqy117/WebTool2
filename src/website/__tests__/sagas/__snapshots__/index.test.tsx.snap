// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sagas/index rootSaga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "REQUEST_BUILD",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`sagas/index rootSaga 2`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "RECEIVE_BUILD",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
